### DISCUSSION TITLE
Taper in time-dependent

### DISCUSSION THREAD

**Discussion by mgracia9** [svenreiche/Genesis-1.3-Version4#160] [Q&A]
Hello, I am trying to run a time-dependent simulation with a taper undulator..I defined the tapering by defining a line with different undulator elements  each with different aw value. I was able to recreate my older Genesis-1.3v2 results in time-independent mode but in time-dependent mode I am struggling on finding power enhancement, any suggestions?

**Comment by ZeugAusHH** [comment]
Hi,
could you provide a minimum working example?
Best
Christoph

**Reply by mgracia9** [reply]
&setup
rootname=CH_3
lattice=example.txt
beamline=ChinU
lambda0= 1.35e-08
gamma0 = 1093.9
delz= 0.036 
shotnoise=true
npart= 8192
nbins = 8
beam_global_stat = true
field_global_stat = true
write_semaphore_file=true
seed = 1
&end

&lattice
zmatch=0.18
&end

&time
time=true
slen = 0.000006
sample=2
s0=0.0
&end

&profile_gauss
label = gauss
c0 = 3800
s0 = 0.00000 
sig = 1.8e-6 
&end

&field
power= 0
dgrid=0.00050512 
ngrid=261
#waist_size= 46.40437889e-6
&end

&beam
current=@gauss
delgam= 2.48 
ex= 6.77e-7 
ey= 6.77e-7 
#betax= 3.1139955968 
#betay= 1.7066940547 
alphax=0
alphay=0
&end

&track
zstop=34 
output_step=1
beam_dump_step = 25
&end

&write
beam=SB_beam3
&end

**Reply by mgracia9** [reply]
[example.txt](https://github.com/svenreiche/Genesis-1.3-Version4/files/14667775/example.txt)

The lattice example. Thank you.

**Comment by svenreiche** [comment]
Thanks, I will have a look into it. Unfortunately I am soon for two weeks on vacation and can do it only after it.

Best
Sven
